lib/openhab/dsl/items/semantics.rb

```diff
@@ -0,0 +1,228 @@
+      #   Note that when comparing against semantic tags, it does a sub-class check.
+      #   So if you search for [Control], you'll get items tagged with [Switch].
+      # @return [Array<GenericItem>]
+      def points(*point_or_property_types)
+        # automatically search the parent equipment (or location?!) for sibling points
+        unless equipment? || location?
+          result = (equipment || location)&.points(*point_or_property_types) || []
+          # remove self. but avoid state comparisons
+          result.delete_if { |item| item.eql?(self) }
+          return result
+        end
+
+        members.points(*point_or_property_types)
+      end
+    end
+
+    GenericItem.include(Semantics)
+  end
+end
+
+# Additions to Enumerable to allow easily filtering groups of items based on the semantic model
+module Enumerable
+  # Returns a new array of items that are a semantics Location (optionally of the given type)
+  def sublocations(type = nil)
+    raise ArgumentError, 'type must be a subclass of Location' if type && !(type < OpenHAB::DSL::Semantics::Location)
+
+    result = select(&:location?)
+    result.select! { |i| i.location_type <= type } if type
+
+    result
+  end
+
+  # Returns a new array of items that are a semantics equipment (optionally of the given type)
+  #
+  # @example Get all TVs in a room
+  #   lGreatRoom.equipments(Semantics::Screen)
+  def equipments(type = nil)
+    raise ArgumentError, 'type must be a subclass of Equipment' if type && !(type < OpenHAB::DSL::Semantics::Equipment)
+
+    result = select(&:equipment?)
+    result.select! { |i| i.equipment_type <= type } if type
+
+    result
+  end
+
+  # Returns a new array of items that are semantics points (optionally of a given type)
+  #
+  # @example Get all the power switch items for every equipment in a room
+  #   lGreatRoom.equipments.flat_map(&:members).points(Semantics::Switch)
+  def points(*point_or_property_types) # rubocop:disable Metrics
+    unless (0..2).cover?(point_or_property_types.length)
+      raise ArgumentError, "wrong number of arguments (given #{point_or_property_types.length}, expected 1..2)"
+    end
+    unless point_or_property_types.all? do |tag|
+             tag < OpenHAB::DSL::Semantics::Point || tag < OpenHAB::DSL::Semantics::Property
+           end
+      raise ArgumentError, 'point_or_property_types must all be a subclass of Point or Property'
+    end
+    if point_or_property_types.count { |tag| tag < OpenHAB::DSL::Semantics::Point } > 1 ||
+       point_or_property_types.count { |tag| tag < OpenHAB::DSL::Semantics::Property } > 1
+      raise ArgumentError, 'point_or_property_types cannot both be a subclass of Point or Property'
+    end
+
+    select do |point|
+      next unless point.point?
+
+      point_or_property_types.all? do |tag|
+        (tag < OpenHAB::DSL::Semantics::Point && point.point_type <= tag) ||
+          (tag < OpenHAB::DSL::Semantics::Property && point.property_type <= tag)
```

Comment:
I came across a situation where point.property_type is nil, so this line raised an error `semantics.rb at line 227: (NoMethodError) undefined method `<=' for nil:NilClass`

```suggestion
          (tag < OpenHAB::DSL::Semantics::Property && point.property_type&.<=(tag))
```

Perhaps this would also apply to the previous line in the case where point_type could return nil.